plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2356,8 +2360,12 @@ public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
         }
 
         String dependencies = sourceTableHandles.stream()
-                .map(handle -> (IcebergTableHandle) handle)
-                .map(handle -> handle.getSchemaTableName() + "=" + handle.getSnapshotId().map(Object.class::cast).orElse(""))
+                .map(handle -> {
```

Comment:
Currently the freshness is considered based on Iceberg tables.
However historization of the changes is done for Delta tables as well.
Just something to consider doing at the engine level.

Reply:
Very good point. This might get addressed in https://github.com/trinodb/trino/pull/12499